elastica/rod/cosserat_rod.py

```diff
@@ -520,6 +520,164 @@ def compute_shear_energy(self):
             * (_batch_dot(sigma_diff, shear_internal_forces) * self.rest_lengths).sum()
         )
 
+    def compute_position_of_point(self, point, index):
+        """
+        Computes the position in the inertial frame of a point specified in the local frame of
+        the specified node of the rod.
+
+        Parameters
+        ----------
+        point : np.array
+            1D (3,) numpy array containing 'float' data.
+            The point describes a position in the local frame relative to the inertial position of node
+            with index `index` and orientation of element with `index`.
+        index: int
+            Index of the node / element in the rod.
+
+        Returns
+        -------
+        position : np.array
+            1D (3,) numpy array containing 'float' data.
+            Position of the point in the inertial frame.
+
+        Examples
+        --------
+        Compute position in inertial frame for a point (0, 0, 1) relative to the last node of the rod.
+
+        >>> rod.compute_position_of_point(np.array([0, 0, 1]), -1)
+        """
+        assert point.shape == (3,), "Point must be in the shape (3,)"
+        position = self.position_collection[..., index] + np.dot(
```

Comment:
I still need to look PR, but why do you define the point as an actual point in space instead of some percentage of lengths. Since rod element lengths can change due to extension.

Reply:
@armantekinalp Thanks for your question. These points are always in the local coordinate system relative to a node. So when the rod elongates, this point moves with the node. The main idea for usage is for the ends of the rods: the joint might not be attached along the backbone of the rod, but rather touching the cylindrical surface. Or the joint might be displaced by a small distance from the end of a rod through an (unmodelled) rigid element.

Reply:
@armantekinalp  thoughts?